drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/SubHeaderBuilder.java

```diff
@@ -105,6 +128,99 @@ public void build() {
+            if (brlConditionVariableColumn.getVarName().isEmpty()) {
+                list.add(new DTCellValue52(true));
+            } else {
+                String varName = brlConditionVariableColumn.getVarName();
+                String key = "";
+                if (!varName.startsWith("$")) {
+                    key = varName;
+                    varName = "@{" + varName + "}";
+                } else {
+                    key = Integer.toString(i);
+                    varName = "@{" + i + "}";
+                }
+                varListInOrder.put(brlConditionVariableColumn.getVarName(), key);
+                list.add(new DTCellValue52(varName));
+            }
+        }
+        dt.getData().add(list);
+
+        String marshal = subString(GuidedDTDRLPersistence.getInstance().marshal(dt), "when", "then");
+
+        int varIndex = 1;
+        for (BRLConditionVariableColumn childColumn : brlColumn.getChildColumns()) {
+
+            final String var = varListInOrder.get(childColumn.getVarName());
+
+            final Pattern pattern = Pattern.compile("@\\{(" + var + ")}");
+            final Matcher matcher = pattern.matcher(marshal);
+            marshal = matcher.replaceAll("\\$" + varIndex++);
+        }
+
+        makeHeaderAndTitle(CONDITION,
+                           brlColumn.getHeader());
+        fieldRow.createCell(targetColumnIndex).setCellValue(marshal);
+    }
+
+    private String subString(String marshal,
+                             final String from,
+                             final String to) {
+        marshal = marshal.substring(marshal.indexOf(from) + from.length());
+        marshal = marshal.substring(0, marshal.lastIndexOf(to));
+
+        marshal = Arrays.stream(marshal.split("\\r?\\n")).map(String::trim).collect(Collectors.joining("\n"));
+        return marshal;
+    }
+
+    private void makeBRLActionColumn(final BRLActionColumn brlColumn) {
+
+        final GuidedDecisionTable52 dt = new GuidedDecisionTable52();
+        dt.getConditions().addAll(dtable.getConditions());
+        final ArrayList<DTCellValue52> list = new ArrayList<>();
+
+        for (IAction iAction : brlColumn.getDefinition()) {
+            if (iAction instanceof ActionInsertFact) {
+                ActionInsertFact insertFact = (ActionInsertFact) iAction;
+                if (insertFact.getBoundName() == null) {
+                    insertFact.setBoundName("brlColumnFact" + BRLActionColumnCount++);
+                }
+            }
+        }
+
+        for (int i = 0; i < dt.getExpandedColumns().size(); i++) {
+            list.add(dtable.getData().get(0).get(i));
+        }
+
+        for (int i = 0; i < brlColumn.getChildColumns().size(); i++) {
+            if (brlColumn.getChildColumns().get(i).getVarName().isEmpty()) {
+                list.add(new DTCellValue52(true));
+            } else {
+                list.add(new DTCellValue52("$" + (i + 1)));
```

Comment:
Shouldn't we do same as for conditions?
```
  if (!varName.startsWith("$")) {
                    key = varName;
                    varName = "@{" + varName + "}";
                } else {
                    key = Integer.toString(i);
                    varName = "@{" + i + "}";
                }
```

Reply:
I'll try with a more complex test file.

Reply:
It did need this. I improved the test file to cover this and added the feature.